src/In.ProjectEKA.TMHHip/DataFlow/Collect.cs

```diff
@@ -0,0 +1,752 @@
+        }
+
+        private static EndpointResource GetEndpointResource(DiagnosticReportAsImage reportAsImage,
+            string endpointId, Text repImagingStudyText)
+        {
+            var connectionType =
+                new ConnectionType("http://terminology.hl7.org/CodeSystem/endpoint-connection-type",
+                    "dicom-wado-rs");
+            return new EndpointResource(HiType.Endpoint, endpointId, repImagingStudyText, "active",
+                connectionType, new List<PayloadType> {new PayloadType(reportAsImage.PayloadType)},
+                new List<string> {reportAsImage.PayloadMimeType},
+                reportAsImage.ReportUrl);
+        }
+
+        private static ImagingStudyResource GetImagingStudyResource(string endpointId,
+            DiagnosticReportAsImage reportAsImage, Text repImagingStudyText, string imagingStudyId)
+        {
+            var subjectImagingStudy = new SubjectDiagReport(reportAsImage.SubjectReference);
+            return new ImagingStudyResource(HiType.ImagingStudy, imagingStudyId,
+                repImagingStudyText, subjectImagingStudy, "available", reportAsImage.StudyStartDate,
+                reportAsImage.NumberOfSeries,
+                reportAsImage.NumberOfInstances,
+                new List<Endpoint> {new Endpoint("Endpoint" + "/" + endpointId)});
+        }
+
+        private static DiagnosticReportImage GetDiagRepResource(DiagnosticReportAsImage reportAsImage,
+            string imagingStudyId, string id, string patientName)
+        {
+            var repText = new Text("additional",
+                "<div xmlns=\"http://www.w3.org/1999/xhtml\">Unstructured data can be sent here</div>");
+            var imagingStudy = new DiagnosticReportImagingStudy("ImagingStudy" + "/" + imagingStudyId);
+            return new DiagnosticReportImage(HiType.DiagnosticReport, id, repText, "final",
+                new Code(reportAsImage.ReportText), new Subject(patientName), reportAsImage.EffectiveDateTime,
+                reportAsImage.Issued, new List<Performer> {new Performer(reportAsImage.Performer)},
+                new List<DiagnosticReportImagingStudy> {imagingStudy},
+                reportAsImage.ReportConclusion);
+        }
+
+        private static async Task<Option<DiagnosticReportResponse>> FetchDiagnosticReportPdfsData(
+            TraceableDataRequest dataRequest, List<DiagnosticReportAsPdf> diagnosticReportAsPdfs, string patientName)
+        {
+            LogDataRequest(dataRequest);
+            var representations = new List<IDiagnosticReport>();
+            foreach (var reportAsPdf in diagnosticReportAsPdfs)
+            {
+                if (!WithinRange(dataRequest.DateRange, reportAsPdf.Issued))
+                {
+                    continue;
+                }
+
+                var uuid = Uuid.Generate().Value;
+                var id = uuid.Split(":").Last();
+                var pdfText =
```

Comment:
Codacy found an issue: [Remove the unused local variable 'pdfText'.](https://app.codacy.com/gh/ProjectEKA/hip-service/pullRequest?prid=4940156)